numba/cuda/tests/cudapy/test_intrinsics.py

```diff
@@ -469,25 +477,60 @@ def test_hdiv_scalar(self):
         np.testing.assert_allclose(ary[0], ref)
 
     @skip_unless_cc_53
-    def test_hdiv_scalar2(self):
-        compiled = cuda.jit("void(f2[:], f2, f2)")(simple_hdiv_scalar)
-        ary = np.zeros(1, dtype=np.float16)
+    def test_hdiv(self):
+        compiled = cuda.jit("void(f2[:], f2[:], f2[:])")(simple_hdiv_kernel)
         arry1 = np.random.randint(-65504, 65505, size=500).astype(np.float16)
         arry2 = np.random.randint(-65504, 65505, size=500).astype(np.float16)
-        for arg1, arg2 in np.nditer([arry1, arry2]):
-            compiled[1, 1](ary, arg1, arg2)
-            ref = arg1 / arg2
-            np.testing.assert_allclose(ary[0], ref)
+        ary = np.zeros_like(arry1, dtype=np.float16)
+
+        compiled.forall(ary.size)(ary, arry1, arry2)
+        ref = arry1 / arry2
+        np.testing.assert_allclose(ary, ref)
 
     @skip_unless_cc_53
-    def test_hdiv_unique_reg_predicate(self):
-        compiled = cuda.jit("void(f2[:], f2, f2)")(simple_hdiv_unique)
-        ary = np.zeros(1, dtype=np.float16)
-        arg1 = np.float16(3.1415926)
-        arg2 = np.float16(1.57)
-        compiled[1, 1](ary, arg1, arg2)
-        ref = (arg1 / arg2) + (arg2 / arg1)
-        np.testing.assert_allclose(ary[0], ref)
+    def test_hdiv_denormal(self):
+        compiled = cuda.jit("void(f2[:], f2[:], f2[:])")(simple_hdiv_kernel)
+
+        # This test is designed to test divison of fp16 constants that
+        # result in small denormal numbers that should not be further
+        # iterated using the Newton Raphson algorithm used in the hdiv
```

Comment:
It appears that `arry1 / arry2` is:

```
array([0.004738, 0.00912 , 0.010574, 0.0164  , 0.01787 , 0.01932 ,
       0.02078 , 0.02223 , 0.03244 , 0.0339  , 0.03537 , 0.03683 ,
       0.03827 , 0.03973 , 0.0412  , 0.04266 , 0.04413 , 0.04556 ,
       0.003685, 0.00841 , 0.009   , 0.00958 , 0.01548 , 0.0173  ,
       0.01843 , 0.01875 , 0.02008 , 0.02138 , 0.02727 , 0.03021 ,
       0.03116 ], dtype=float16)
```

Where do the small denormals appear?

Reply:
The denormal being checked occurs after the first iteration of the Newton Raphson (NR) algorithm; if a denormal is found the second iteration of the NR algorithm is skipped